src/encoded/static/components/filegallery.js

```diff
@@ -3256,26 +3350,97 @@ FileGalleryRendererComponent.contextTypes = {
 const FileGalleryRenderer = auditDecor(FileGalleryRendererComponent);
 
 
-const CollapsingTitle = (props) => {
-    const { title, handleCollapse, collapsed } = props;
+const CollapsingTitle = (props, reactContext) => {
+    const { title, handleCollapse, collapsed, context, filters, analysisObjectKey, outputCategory, outputType, totalFiles, isDownloadable } = props;
+
+    const [downloadModalVisibility, setDownloadModalVisibility] = useState(false);
+
+    const handleDownloadClick = () => {
+        const { accession } = context;
+        const type = context && context['@type'] ? context['@type'][0] : 'Experiment';
+        let analysisObjectId = '';
+
+        if (context.analysis_objects) {
+            const analysisObjects = context.analysis_objects.find((analysisObject) => analysisObject.accession === analysisObjectKey);
+
+            if (analysisObjects) {
+                analysisObjectId = analysisObjects['@id'];
+            }
+        }
+
+        const fileTypes = filters.file_type && filters.file_type.length > 0
+            ? filters.file_type.map((fileType) => `&files.file_type=${encodedURIComponent(fileType)}`).join('')
+            : '';
+        const biologicalReplicates = filters.biological_replicates && filters.biological_replicates.length > 0
+            ? filters.biological_replicates.map((replicate) => `&files.biological_replicates=${encodedURIComponent(replicate)}`).join('')
+            : '';
+        const outputTypes = filters.output_type && filters.output_type.length > 0
+            ? filters.output_type.map((o) => `&files.output_type=${encodedURIComponent(o)}`).join('')
+            : '';
+        const fileStatus = document.querySelector('[name="filterIncArchive"]').checked
```

Comment:
Should pass the `inclusionOn` state here. Again, I realize this crosses components here, but this bypasses React’s state management, so I’d consider this non-optional.

Reply:
Done. Thanks again. for the catch. Using document.querySelector() is very addictive.